### ISSUE TITLE
Add keyboard navigation to repeater component

### ISSUE THREAD

**Issue by Blackbaud-AlexKingman** [blackbaud/skyux-lists#60]
When tabbing into list, first list item will be focused. At this point, the arrow controls will take effect:

Down arrow key: set focus on next list item (if available).
Up arrow key: set focus on previous list item (if available).
Right arrow key: set focus on first interactive child element, and then continue to iterate through them if right arrow is pressed again.
Left arrow key: set focus on previous interactive child element. If focus has reached the first interactive child element, set focus back on list item.
While focused on a list item, hitting the tab key will move focus out of the list and to the next focusable element on the page.

**Comment by Blackbaud-AlexKingman** [comment]
When tabbing into list, first list item will be focused. At this point, the arrow controls will take effect:
- Down arrow key: set focus on next list item (if available).
- Up arrow key: set focus on previous list item (if available).
- Right arrow key: set focus on first interactive child element, and then continue to iterate through them if right arrow is pressed again.
- Left arrow key: set focus on previous interactive child element. If focus has reached the first interactive child element, set focus back on list item.

While focused on a list item, hitting the tab key will move focus out of the list and to the next focusable element on the page.

For select-mode, the checkbox will no longer be focussable, and instead be selected by using the enter key or space bar while the list item is focused.
